integration-test/playwright/harmful-apis.spec.js

```diff
@@ -0,0 +1,140 @@
+import { test, expect } from '@playwright/test'
+import { readFileSync } from 'fs'
+import { mockWindowsMessaging, wrapWindowsScripts } from '@duckduckgo/messaging/lib/test-utils.mjs'
+import { perPlatform } from './type-helpers.mjs'
+
+test('Harmful APIs protections', async ({ page }, testInfo) => {
+    const protection = HarmfulApisSpec.create(page, testInfo)
+    const results = await protection.enabled();
+    // note that if protections are disabled, the browser will show a device selection pop-up, which will never be dismissed
+
+    [
+        'deviceOrientation',
+        'GenericSensor',
+        'UaClientHints',
+        'NetworkInformation',
+        'getInstalledRelatedApps',
+        'FileSystemAccess',
+        'WindowPlacement',
+        'WebBluetooth',
+        'WebUsb',
+        'WebSerial',
+        'WebHid',
+        'WebMidi',
+        'IdleDetection',
+        'WebNfc',
+        'StorageManager'
+    ].forEach((name) => {
+        for (const result of results[name]) {
+            expect(result.result).toEqual(result.expected)
+        }
+    })
+})
+
+export class HarmfulApisSpec {
+    htmlPage = '/harmful-apis/index.html'
+    config = './integration-test/test-pages/harmful-apis/config/apis.json'
+
+    /**
+     * @param {import("@playwright/test").Page} page
+     * @param {import("./type-helpers.mjs").Build} build
+     * @param {import("./type-helpers.mjs").PlatformInfo} platform
+     */
+    constructor (page, build, platform) {
+        this.page = page
+        this.build = build
+        this.platform = platform
+    }
+
+    async enabled () {
+        await this.installPolyfills()
+        const config = JSON.parse(readFileSync(this.config, 'utf8'))
+        await this.setup({ config })
+        await this.page.goto(this.htmlPage)
+        for (const button of await this.page.getByTestId('user-gesture-button').all()) {
+            await button.click()
+        }
+        const resultsPromise = this.page.evaluate(() => {
+            return new Promise(resolve => {
+                window.addEventListener('results-ready', () => {
+                    // @ts-expect-error - this is added by the test framework
+                    resolve(window.results)
+                })
+            })
+        })
+        await this.page.getByTestId('render-results').click()
+        return await resultsPromise
+    }
```

Comment:
can you separate these actions+assertions from the 'enabled' call (which is just the setup) - just inline it inside the `test` above, or create a separate method on this class. 

The examples in the duckplayer.spec.js are the format we want to move forward with, but since this is a single test I don't think it's an issue to just inline the actions/assertions into the `test` <- that's what I did in the windows-permissions tests